### PR TITLE
Implement timeout for socks connection

### PR DESCRIPTION
**Pull request by RCasatta**
To achieve this we have to include a modified socks library, with supports for
timeout

Remove the error `BothSocksAndTimeout` which isn't needed anymore

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -23,7 +23,7 @@ serde = { version = "^1.0", features = ["derive"] }
 serde_json = { version = "^1.0" }
 
 # Optional dependencies
-socks = { version = "0.3", optional = true }
```
Comment (afilini) [inline Cargo.toml:26]: I don't think I can publish to crates.io with a local dependency. I think you need to move `socks` inside our `src` folder and include it as a local module

Comment (RCasatta) [reply Cargo.toml:26]: ahah ooops

Comment (RCasatta) [reply Cargo.toml:26]: `socks` included in `src`

**Comment by RCasatta** [comment]
not necessary anymore

**Comment by RCasatta** [comment]
I closed the wrong one

**Pull request merged**